### DISCUSSION TITLE
Typescript Types don't seem to align with docs in version 7.8.6 for useFieldArray

### DISCUSSION THREAD

**Discussion by ckeyes88** [react-hook-form/react-hook-form#5621] [Q&A]
**Describe the bug**
I've recently updated to the latest version (v7.8.6) and have been struggling to get typings to compile properly. I assumed it was something to do with my set up but when I fork the code sandbox and add the latest version I get the same (and actually more) type errors that I can't resolve.

**To Reproduce**
I saved the sandbox with no code changes other than updating the `react-hook-form` to 7.8.6. As you can see loads properly on the right but crashes when you try to actually edit the form.

**Codesandbox link (Required)**
[Sandbox](https://codesandbox.io/embed/react-hook-form-usefieldarray-forked-noqrl?fontsize=14&hidenavigation=1&theme=dark)

**Expected behavior**

**Screenshots**
[Here's a GIF](https://share.getcloudapp.com/jkuPYKZp) showing the code and form behavior: https://share.getcloudapp.com/jkuPYKZp

**Additional context**
- Typescript version 3.3.3, 4.2.4 and 4.3.2 all behave the same.
- I eventually got my code to compile by doing `defaultValue={(field as {id: string, value: string}).value}` which isn't ideal.

**Comment by bluebill1049** [comment]
> As you can see loads properly on the right but crashes when you try to actually edit the form.

provide more details and steps to reproduce, where is edit? also input is not registered as well.

**Reply by MatMath** [reply]
@ckeyes88 I think you are missing the definition on `useForm<>` from the Sandbox code if you add the proper type for the data structure it should work:
```
interface Thing {
  label: string,
  dateRange: {
    startsAt: string,
    endsAt: string,
  }
}
interface FormField {
  things: Thing[]
}

with the proper typing: 
... = useForm<FormField>({ ...
```

When you use the field you need to use the proper key so:
```
<input
    name={`things[${index}].label`}
    defaultValue={item.value}    > this should be item.label since this is the proper type from Thing
  />

```

You do still need to add typing in the button like: 
```
setValue(`things.${index}.label` as `things.0.label`, "00:00", {
```
